cmd/syncer/main.go

```diff
@@ -156,43 +170,93 @@ func startSyncers(ctx context.Context, GVRs []string, secretRef client.ObjectKey
 	if err != nil {
 		return nil, fmt.Errorf("unable to create cluster config from control plane secret: %v", err)
 	}
-	upstreamRestConfig, err := clusterSecret.RestConfigFromSecret(controlPlaneSecret)
-	if err != nil {
-		return nil, fmt.Errorf("unable to create rest config from control plane secret: %v", err)
-	}
 	upstreamClient, err := clusterSecret.DynamicClientsetFromSecret(controlPlaneSecret)
 	if err != nil {
 		return nil, fmt.Errorf("unable to create client from control plane rest config: %v", err)
 	}
 	informerFactory := dynamicinformer.NewDynamicSharedInformerFactory(upstreamClient, DEFAULT_RESYNC)
-	logger.Info("starting informer")
 	informerFactory.Start(ctx.Done())
-	logger.Info("waiting for cache sync")
 	informerFactory.WaitForCacheSync(ctx.Done())
-	logger.Info("cache sync complete")
 
+	//for spec: upstream is control-plane, downstream is data plane
 	specSyncConfig := syncer.Config{
-		UpstreamClientConfig: upstreamRestConfig,
-		GVRs:                 GVRs,
-		DownStreamClient:     mgr.GetClient(),
-		InformerFactory:      informerFactory,
-		ClusterID:            upstreamClusterConfig.Name,
-		NeverSyncedGVRs:      NEVER_SYNCED_GVRs,
+		GVRs:            GVRs,
+		InformerFactory: informerFactory,
+		ClusterID:       upstreamClusterConfig.Name,
```

Comment:
the cluster id where does that come from?

Reply:
it's the name data field in the cluster config secret.